test/apps/iodemo/io_demo.cc

```diff
@@ -1975,6 +1977,16 @@ class DemoClient : public P2pDemoCommon {
             LOG << "invalid address " << server_addr;
             abort();
         }
+        if (opts().src_addr != NULL) {
+            memset(&local_addr, 0, sizeof(local_addr));
+            local_addr.sin_family = AF_INET;
```

Comment:
Can we use a sperate function to unify lines 1981~1987 with server_addr code?

Reply:
called existing func ucs_sock_ipstr_to_sockaddr in the two places.